### PR TITLE
[AMDGPU] SILowerI1Copies: clear kill flags on COPY

### PR DESCRIPTION
**Pull request by perlfu**
Clear kill flags on COPY source as it will be reused.

### CODE REVIEW

File: llvm/test/CodeGen/AMDGPU/lower-i1-copies-clear-kills.mir [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+  ; CHECK-NEXT:   S_ENDPGM 0
+  bb.0:
+    successors: %bb.1(0x80000000)
+    liveins: $vgpr0, $vgpr1
+
+    %17:vgpr_32 = COPY $vgpr1
+    %16:vgpr_32 = COPY $vgpr0
+    %19:sreg_32 = S_MOV_B32 0
+    %20:sreg_32 = V_CMP_NE_U32_e64 %16, %19, implicit $exec
+    %21:sgpr_32 = S_MOV_B32 0
```
Comment (arsenm) [inline llvm/test/CodeGen/AMDGPU/lower-i1-copies-clear-kills.mir:91]: Can you run this through -run-pass=none to compact the register numbers

File: llvm/lib/Target/AMDGPU/SILowerI1Copies.cpp [hunk 2]
```diff
@@ -708,6 +708,9 @@ bool SILowerI1Copies::lowerCopiesToI1() {
             .addImm(0);
         MI.getOperand(1).setReg(TmpReg);
         SrcReg = TmpReg;
+      } else {
+        // SrcReg needs to be live beyond copy.
+        MI.clearKillInfo();
```
Comment (arsenm) [inline llvm/lib/Target/AMDGPU/SILowerI1Copies.cpp:713]: Don't you know the exact operand?

File: llvm/test/CodeGen/AMDGPU/lower-i1-copies-clear-kills.mir (hunk 1 above)
Comment (perlfu) [reply llvm/test/CodeGen/AMDGPU/lower-i1-copies-clear-kills.mir]: Done

File: llvm/lib/Target/AMDGPU/SILowerI1Copies.cpp (hunk 2 above)
Comment (perlfu) [reply llvm/lib/Target/AMDGPU/SILowerI1Copies.cpp]: Done

**Review by arsenm — APPROVED** [review]

**Pull request merged**